src/com/linkedin/parseq/Engine.java

```diff
@@ -103,6 +96,37 @@
     _allLogger = loggerFactory.getLogger(LOGGER_BASE + ":all");
     _rootLogger = loggerFactory.getLogger(LOGGER_BASE + ":root");
 
+    if (_properties.containsKey(MAX_RELATIONSHIPS_PER_TRACE)) {
+      _maxRelationshipsPerTrace = (Integer) getProperty(MAX_RELATIONSHIPS_PER_TRACE);
+    } else {
+      _maxRelationshipsPerTrace = DEFUALT_MAX_RELATIONSHIPS_PER_TRACE;
+    }
+
+    if (_properties.containsKey(MAX_CONCURRENT_PLANS)) {
+      _maxConcurrentPlans = (Integer) getProperty(MAX_CONCURRENT_PLANS);
+    } else {
+      _maxConcurrentPlans = DEFUALT_MAX_CONCURRENT_PLANS;
+    }
+    _concurrentPlans = new Semaphore(_maxConcurrentPlans);
```

Comment:
I would leave it as an implementation detail. I see two scenarios:
1) In setting like kafka stream consumer the submitting thread is expecting to block and in this case there will be no timeout
2) In async settings, if Engine is above it's capacity some task submissions will potentially time out but it is not clear to me why one would expect FIFO semantics, e.g. is FIFO better than other strategy, if so, why?